spec/requests/api/v1/communications/show_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Communications show', type: :request do
+  let!(:user) { create(:user) }
+  let!(:auth_headers) { user.create_new_auth_token }
+  let!(:communication) { create(:communication, published: true) }
+  let!(:unpublished_communication) { create(:communication, published: false) }
+
+  describe 'GET api/v1/communications/<id>' do
+    context 'with authentication' do
+      it 'shows the communication if published' do
+        get "/api/v1/communications/#{communication.id}", headers: auth_headers
+        expect(response).to have_http_status 200
+        response_body = Oj.load(response.body)['communication']
+        serialized_communication = communication.as_json
+        expect(response_body['id']).to eq(serialized_communication['id'])
+        expect(response_body['title']).to eq(serialized_communication['title'])
+        expect(response_body['text']).to eq(serialized_communication['text'])
+        expect(response_body['updated_at']).to eq(serialized_communication['updated_at'])
```

Comment:
expect(response_body['id']).to eq(communication.as_json(only: %i[id title text updated_at ]))